### PR TITLE
Unbond recovery

### LINKED ISSUE #1703: Allow users to self rectify unsatisfiable unbondings
## Summary

Periodically, an unbonding may be complete and for some reason we have been able to pin down, once the unbonding is complete, we are unable to fully satisfy the users' unbonding request - when the MsgSend is executed the funds are not there having been rebonded. This is possibly down to a race condition between the DelegationFlush logic running before the MsgSend is executed.

Currently this requires developer intervention in the form of a upgrade handler to requeue the unbonding or release the escroewed qassets.

Adding a transaction type to allow users, under the right circumstances, to cancel or requeue an unbonding.

**SendErrors** - add a new field of type int64 to the WithdrawalRecord struct, to track the number of times the MsgSend has been attempted and failed. Default is 0.

**MsgCancelUnbonding**

_MsgCancelQueuedUnbonding_ already exists (although not yet added to the front end), and we should rename and extend the capabilities of this message type to allow users to cancel an unbonding in the state QUEUED (as before) but additionally in the state UNBONDING with SendErrors > 0. This action deletes the unbonding record and returns the qassets from escrow to the  user.

This message comprises the following fields, with the following validity criteria:
**chainId** - must be a valid chain id for a supported zone
**hash** - must be the hash of an existing withdrawalRecord in QUEUED or UNBONDING state. If the latter, SendErrors field must be greater than 0.
**fromAddress** - message signer. Must either match the withdrawalRecord.Delegator field of the withdrawalRecord matching hash, OR the governance account. 

**MsgRequeueUnbonding**

A new message should be added to allow users to requeue their unbonding, in the event that is has state UNBONDING, and SendErrors > 0. 

This message comprises the following fields, with the following validity criteria:
**chainId** - must be a valid chain id for a supported zone
**hash** - must be the hash of an existing withdrawalRecord in UNBONDING state with SendErrors field must be greater than 0.
**fromAddress** - message signer. Must either match the withdrawalRecord.Delegator field of the withdrawalRecord matching hash, OR the governance account.

### PR DESCRIPTION
**Pull request by joe-bowman**
## 1. Summary
Fixes #1703

- Add SendErrors int64 field to WithdrawalRecord. Defaults to 0.
- Rename MsgCancelQueuedRedemption to MsgCancelRedemption; update functionality to also apply to STATUS_QUEUED withdrawalRecords with SendErrors > 0.
- Add MsgRequeueRedemption to set the status of a withdrawalRecord in STATUS_UNBONDING with SendErrors > 0 to STATUS_QUEUED (and resetting SendErrors)
- 
## 2.Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## 3. TODO

- [ ] Add CLI for new message types

## Summary by CodeRabbit

- **New Features**
	- Added functionality to cancel and requeue failed redemption requests via CLI commands.
	- Introduced a new API endpoint for requeuing redemptions.
	- Enhanced API documentation for clarity on redemption processes.

- **Bug Fixes**
	- Improved error tracking for send operations with the addition of a new field in data structures.

- **Documentation**
	- Updated API documentation to reflect changes in endpoint functionalities and response types. 

- **Chores**
	- Updated the version of protocol buffers used in the project.
	- Enhanced security checks for redemption cancellation and requeuing processes.

### CODE REVIEW

File: x/interchainstaking/keeper/msg_server_test.go [hunk 1]
```diff
@@ -933,6 +986,161 @@
...
+			"no zone exists",
+			func(s *KeeperTestSuite) *icstypes.MsgRequeueRedemption {
+				return &icstypes.MsgRequeueRedemption{
+					ChainId:     "bob",
+					Hash:        hash,
+					FromAddress: addressutils.GenerateAddressForTestWithPrefix("quick"),
+				}
+			},
+			fmt.Sprintf("no unbonding record with hash \"%s\" found", hash),
+		},
```
[the remark this answers is not in the document]
Comment (joe-bowman) [reply x/interchainstaking/keeper/msg_server_test.go:1006]: The field here is a string, not an error.

**Review by faddat — APPROVED** [review]
nice :D

**Comment by faddat** [comment]
I think coderabbit is harassing you for UK spelling. 

So I will not concern myself with its feedback this time

File: x/interchainstaking/keeper/msg_server.go [hunk 2]
```diff
@@ -122,12 +130,112 @@ func (k msgServer) CancelRedemption(goCtx context.Context, msg *types.MsgCancelQ
...
+	record, found := k.GetWithdrawalRecord(ctx, msg.ChainId, msg.Hash, types.WithdrawStatusUnbond)
+	if !found {
+		return nil, fmt.Errorf("no unbonding record with hash \"%s\" found", msg.Hash)
+	}
+	if record.SendErrors == 0 {
+		return nil, fmt.Errorf("cannot requeue unbond \"%s\" with no errors", msg.Hash)
+	}
+
+	if record.Delegator != msg.FromAddress && k.Keeper.GetGovAuthority(ctx) != msg.FromAddress {
+		return nil, fmt.Errorf("incorrect user for record with hash \"%s\"", msg.Hash)
```
Comment (odeke-em) [inline x/interchainstaking/keeper/msg_server.go:155]: In the future @joe-bowman @faddat we can simply use the format specifier "%q" per https://go.dev/play/p/7SJTkPPMyFl

**Pull request merged**